lib/workers/bin/run_single_worker.rb

```diff
@@ -60,13 +56,13 @@
   create_options = {:pid => Process.pid}
   runner_options = {}
 
-  if ENV["QUEUE"]
-    create_options[:queue_name] = ENV["QUEUE"]
-    runner_options[:ems_id] = worker_class.ems_id_from_queue_name(ENV["QUEUE"]) if worker_class.respond_to?(:ems_id_from_queue_name)
+  if ENV["EMS_ID"].to_i > 0
+    create_options[:queue_name] = "ems_#{ENV['EMS_ID']}"
+    runner_options[:ems_id]     = ENV["EMS_ID"].to_i
```

Comment:
@carbonin Do we actually leverage more than one queue name in the existing QUEUE env variable?  I don't think so since you can't store an array in ENV variables.  Either way, please make sure this is fine.

Reply:
Yes, we need multiple ems_ids for the collapsed workers that @juliancheal worked on.  The ems_id_from_queue_name actually returns a complex string in those cases, so we can't stop calling it

Reply:
https://github.com/ManageIQ/manageiq/blob/481eec1cbaad1259a080f296b6a46c810f989c60/app/models/mixins/per_ems_worker_mixin.rb#L111-L112

Reply:
Oh I can't read...here it is: 
https://github.com/ManageIQ/manageiq/blob/6be82739e07be7ab28a6c71b60ca69585deacf00/app/models/mixins/per_ems_worker_mixin.rb#L111

Reply:
That might be true but it's impossible for `ENV["QUEUE"]` to ever have been an array.  It would have been a string that we pass along to `ems_id_from_queue_name`.  Therefore, I'm suggesting it never supported specifying multiple queue names via the `QUEUE` env variable and my change here would not break anything.

If I'm correct above, we can figure out how to do an array of queue names in a followup PR as it's out of the scope of this one.

Reply:
> it's impossible for ENV["QUEUE"] to ever have been an array

That's true for workers started using spawn, but the idea behind this PR is that some things worked with fork, but not with spawn. I think arrays of queue names may be another of those things.

That said, I'm okay with that as a follow-up. Would probably need some help from @juliancheal to understand how workers with multiple queue names work as I'm having trouble finding where we actually initialize a worker with an array of queues in https://github.com/ManageIQ/manageiq/pull/15711

Reply:
I'm trying to find where we're passing multiple ems_ids in `@cfg`, the only thing I can find was https://github.com/ManageIQ/manageiq-providers-amazon/pull/280/files#diff-c32d8674fab99c608d4b4723ec2ac3f8R10 hopefully @juliancheal can fill in what I'm missing :)

Reply:
https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/per_ems_worker_mixin.rb#L111

Here is where we return to the worker the queue name be it a singular or an array.

Reply:
Thanks @juliancheal !

Reply:
@juliancheal overwrote queue_name accessor to store JSON, so the column would return an Array, which is how it works elsewhere.  However it seems run_single_worker wasn't updated to support that (though it *should* because that's the new entrypoint).

Reply:
@Fryguy Well I overwrote the reader (https://github.com/juliancheal/manageiq/blob/67905a3239eca5a71c30478adc6ebf10239a8924/app/models/miq_worker.rb#L349) to parse JSON, I think postures just magically stores an array of strings into a string by magic.

I'm not sure what `run_single_worker` is, so sorry if I didn't update that too :(

Reply:
I think it's ActiveRecord doing the magic, though I thought it only did that if you said the column was serialized :/

Reply:
Hmm. There be some magic somewhere.

Reply:
Hmm, we need tests in this area then because I had no idea this was a requirement until I looked at `ems_id_from_queue_name` and saw we didn't fully implement an array of queue names.

Note, it's broken before this PR and the prior one so I'm not suggesting we fix it here.

@juliancheal or @agrare can you look at the Procfile.example above, would this make sense for how we'd invoke multiple queue names from command line:

`env EMS_ID=1,2,3 ruby lib/workers/bin/run_single_worker.rb MiqEmsRefreshCoreWorker`

Reply:
@jrafanie I actually tried this earlier but I only got `@cfg[:ems_id] = 1` for some reason I'll have to look into this

Reply:
@agrare yeah, run_single_worker will need to be enhanced to parse the ENV like this:

`runner_options[:ems_id] = ENV['EMS_ID'].split(',').collect(&:to_i)`